app/src/main/java/com/nitramite/adapters/CustomParcelsAdapterV2.java

```diff
@@ -224,8 +227,16 @@ else if (parcelItems.get(position).getParcelCarrier().equals("99") && phase.equa
 
 
         // Fourth line
+        ParcelItem item = parcelItems.get(position);
+        parcelLastMovementStatusTV.setVisibility(!lastUpdate || (item.getLastEventDate() == null || parcelItems.get(position).getArchivedPackage()) ? View.GONE : View.VISIBLE);
         if (!parcelItems.get(position).getArchivedPackage()) {
             parcelUpdateStatusTV.setText(parcelItems.get(position).getParcelUpdateStatus());
+            if (lastUpdate) {
+                if (item.getLastEventDate() != null) {
+                    parcelUpdateStatusTV.setVisibility(View.GONE);
+                    parcelLastMovementStatusTV.setText(context.getString(R.string.last_change, item.getLastEventDate()));
+                }
+            }
```

Comment:
It's very difficult to follow what's going on here. First of all `parcelItems.get(position)` is set to a variable but it's then still used 3 times. If I understood correctly what this does, could this be refactored to something like below? It's more lines but arguably(?) easier to read.

```java
ParcelItem item = parcelItems.get(position);
boolean isArchived = item.getArchivedPackage();
boolean useLastEventDate = showLastEventDate && item.getLastEventDate() != null;
if (!isArchived) {
    parcelUpdateStatusTV.setText(item.getParcelUpdateStatus());
    if (useLastEventDate) {
        parcelUpdateStatusTV.setVisibility(View.GONE);
        parcelLastEventDateTV.setVisibility(View.VISIBLE);
        parcelLastEventDateTV.setText(context.getString(R.string.last_change, item.getLastEventDate()));
    } else {       
        parcelUpdateStatusTV.setVisibility(View.VISIBLE);
        parcelLastEventDateTV.setVisibility(View.GONE);
    }
}
```

Reply:
@norkator comments on this? Would this achieve the same end result? I was not 100% sure as the particular code is quite "convoluted". :grimacing:

Reply:
> It's very difficult to follow what's going on here. First of all `parcelItems.get(position)` is set to a variable but it's then still used 3 times. If I understood correctly what this does, could this be refactored to something like below? It's more lines but arguably(?) easier to read.
> 
> ```java
> ParcelItem item = parcelItems.get(position);
> boolean isArchived = item.getArchivedPackage();
> boolean useLastEventDate = showLastEventDate && item.getLastEventDate() != null;
> if (!isArchived) {
>     parcelUpdateStatusTV.setText(item.getParcelUpdateStatus());
>     if (useLastEventDate) {
>         parcelUpdateStatusTV.setVisibility(View.GONE);
>         parcelLastEventDateTV.setVisibility(View.VISIBLE);
>         parcelLastEventDateTV.setText(context.getString(R.string.last_change, item.getLastEventDate()));
>     } else {       
>         parcelUpdateStatusTV.setVisibility(View.VISIBLE);
>         parcelLastEventDateTV.setVisibility(View.GONE);
>     }
> }
> ```

Hahaha this is my crap again. I need to fix this.